File: __Viejo/Formularios/frm_error_detalle.vb
Public Class frm_error_detalle
    Private Sub frm_error_detalle_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ' Centrar el formulario
        Me.CenterToParent()
    End Sub

    Public Sub MostrarError(titulo As String, mensaje As String, detalles As String)
        Me.Text = titulo
        txt_mensaje.Text = mensaje
        txt_detalles.Text = detalles
        
        ' Seleccionar todo el texto para facilitar la copia
        txt_detalles.SelectAll()
        txt_detalles.Focus()
    End Sub

    Private Sub btn_copiar_Click(sender As Object, e As EventArgs) Handles btn_copiar.Click
        Try
            Dim textoCompleto As String = "TÍTULO: " & Me.Text & vbCrLf & vbCrLf &
                                        "MENSAJE: " & txt_mensaje.Text & vbCrLf & vbCrLf &
                                        "DETALLES: " & txt_detalles.Text
            
            Clipboard.SetText(textoCompleto)
            MessageBox.Show("Error copiado al portapapeles", "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Information)
        Catch ex As Exception
            MessageBox.Show("Error al copiar: " & ex.Message, "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub btn_cerrar_Click(sender As Object, e As EventArgs) Handles btn_cerrar.Click
        Me.Close()
    End Sub

    Private Sub txt_detalles_KeyDown(sender As Object, e As KeyEventArgs) Handles txt_detalles.KeyDown
        ' Ctrl+A para seleccionar todo
        If e.Control AndAlso e.KeyCode = Keys.A Then
            txt_detalles.SelectAll()
            e.Handled = True
        End If
    End Sub
End Class

File: Formularios/frm_error_detalle.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace Centrex
{
    public partial class frm_error_detalle
    {
        public frm_error_detalle()
        {
            InitializeComponent();
        }
        private void frm_error_detalle_Load(object sender, EventArgs e)
        {
            // Centrar el formulario
            CenterToParent();
        }

        public void MostrarError(string titulo, string mensaje, string detalles)
        {
            Text = titulo;
            txt_mensaje.Text = mensaje;
            txt_detalles.Text = detalles;

            // Seleccionar todo el texto para facilitar la copia
            txt_detalles.SelectAll();
            txt_detalles.Focus();
        }

        private void btn_copiar_Click(object sender, EventArgs e)
        {
            try
            {
                string textoCompleto = "TÍTULO: " + Text + Constants.vbCrLf + Constants.vbCrLf + "MENSAJE: " + txt_mensaje.Text + Constants.vbCrLf + Constants.vbCrLf + "DETALLES: " + txt_detalles.Text;

                Clipboard.SetText(textoCompleto);
                MessageBox.Show("Error copiado al portapapeles", "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Information);
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error al copiar: " + ex.Message, "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
        }

        private void btn_cerrar_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void txt_detalles_KeyDown(object sender, KeyEventArgs e)
        {
            // Ctrl+A para seleccionar todo
            if (e.Control && e.KeyCode == Keys.A)
            {
                txt_detalles.SelectAll();
                e.Handled = true;
            }
        }
    }
}
